java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -829,8 +829,7 @@ private void executeAssignement(AssignmentExpressionTree tree) {
     ProgramState.Pop unstack;
     SymbolicValue value;
 
-    boolean isSimpleAssignment = ExpressionUtils.isSimpleAssignment(tree) || ExpressionUtils.isSelectOnThisOrSuper(tree);
-    if (isSimpleAssignment) {
+    if (tree.is(Tree.Kind.ASSIGNMENT)) {
```

Comment:
I have the feeling that it is wrong written like this. In case of `this.x = ` we will unstack too many values, when only the field is on the stack (see how MemberSelects are handled, L#978)
I however agree that using `ExpressionUtils.isSelectOnThisOrSuper(tree);` was probably wrong, as it can be used in operation which are not assignment, and therefore a new SV should be created.

I would go for something slightly different :
```
  private void executeAssignment(AssignmentExpressionTree tree) {
    ProgramState.Pop unstack;
    SymbolicValue value;

    boolean isSimpleAssignment = ExpressionUtils.isSimpleAssignment(tree) || ExpressionUtils.isSelectOnThisOrSuper(tree);
    if (isSimpleAssignment) {
      unstack = programState.unstackValue(1);
      value = tree.is(Tree.Kind.ASSIGNMENT) ? unstack.values.get(0) : constraintManager.createSymbolicValue(tree);
    } else {
      unstack = programState.unstackValue(2);
      value = constraintManager.createSymbolicValue(tree);
    }

    programState = unstack.state;
    programState = programState.stackValue(value);
    if (isSimpleAssignment || tree.variable().is(Tree.Kind.IDENTIFIER) || ExpressionUtils.isSelectOnThisOrSuper(tree)) {
      programState = programState.put(ExpressionUtils.extractIdentifier(tree).symbol(), value);
    }
  }
```